File: src/main/org/jboss/messaging/core/remoting/spi/ConnectorFactory.java
package org.jboss.messaging.core.remoting.spi;

import java.io.Serializable;
import java.util.Map;

/**
 * 
 * A ConnectorFactory
 * 
 * @author <a href="mailto:tim.fox@jboss.com">Tim Fox</a>
 *
 */
public interface ConnectorFactory extends Serializable
{
   Connector createConnector(Map<String, Object> configuration, BufferHandler handler,                           
                             ConnectionLifeCycleListener listener);
}

File: dotnet/JBMClient/remoting/spi/ConnectorFactory.cs
namespace JBoss.JBM.Client.remoting.spi
{

    using System.Collections.Generic;
    
    
    /**
     * 
     * A ConnectorFactory
     * 
     * @author <a href="mailto:tim.fox@jboss.com">Tim Fox</a>
     *
     */
    public interface ConnectorFactory
    {
        Connector createConnector(Dictionary<string, object> configuration, BufferHandler handler,
                                  ConnectionLifeCycleListener listener);
    }
}